pipeline_dp/accumulator.py

```diff
@@ -257,9 +264,34 @@ def compute_metrics(self):
         return dp_computations.add_noise_vector(self._vec_sum, self._params)
 
 
-@dataclass
 class SumParams:
-    noise: dp_computations.MeanVarParams
+    """Parameters for a sum accumulator.
+
+    Wraps epsilon and delta from the budget which are lazily loaded.
+    AggregateParams are copied into a MeanVarParams instance.
+    """
+
+    def __init__(self, budget: pipeline_dp.budget_accounting.MechanismSpec,
+                 aggregate_params: aggregate_params.AggregateParams):
+        self._budget = budget
+        self._aggregate_params = aggregate_params
+
+    @property
+    def eps(self):
+        return self._budget.eps
+
+    @property
+    def delta(self):
+        return self._budget.delta
+
+    @property
+    def mean_var_params(self):
+        return dp_computations.MeanVarParams(
+            self.eps, self.delta, self._aggregate_params.low,
+            self._aggregate_params.high,
```

Comment:
@dvadym 👋 

Not sure if this is exactly right; it doesn't inherit MeanVarParams as a superclass because instantiating it in the `__init__` fails due to the dependency on budget accountant `eps must not be None.` So I created `mean_var_params` as a property so it can be called after compute budgets. 

Also, maybe we want to copy the `aggregate_params` attributes entirely so we don't rely on the other object's state? Is it OK "pythonic" to carry state in this way? 🐍

Reply:
do you mean to have a field in `SumParams` of type `MeanVarParams` ? If yes, it would make sense, but in `__init__` we don't have yet `eps/delta` to construct  `MeanVarParams`.

My feeling is that the current approach is reasonable. And any anyway this code will be refactored when we will introduce using of PldBudgetAccountant.

Reply:
Yes, to copy directly everything except eps/delta which would be more of a stateless/functional style that is not necessary here :) Thanks for clarifying!